host-spec/c06-consensus.tm

```diff
@@ -505,41 +610,35 @@
 
   <subsection|Epoch Randomness><label|sect-epoch-randomness>
 
-  At the end of epoch <math|\<cal-E\><rsub|n>>, each block producer is able
-  to compute the randomness seed it needs in order to participate in the
-  block production lottery in epoch <math|\<cal-E\><rsub|n+2>>. For epoch 0
-  and 1, the randomness seed is provided in the genesis state. The
-  computation of the seed is described in Algorithm
-  <reference|algo-epoch-randomness> which uses the concept of epoch subchain
-  described in Definition <reference|defn-epoch-subchain>.
+  In the beginning of each epoch <math|\<cal-E\><rsub|n>> the host will
```

Comment:
This paragraph is not clear and it seems it is not correct. Why does the host have to receive R_E_{n+1} in the beginning of E_n to participate E_n+1. In the beginning of E_n+1, the host has to obtain the randomness using the VRF outputs belonging to the probabilistically finalized blocks of E_{n-1}.  The previous version was correct.

Reply:
As this spec is only concerned with the host side of things and both the authority set and epoch randomness are given to the host in the same consensus message, this description should be correct.

Reply:
I'm curious to know how the runtime is computing the randomness, are the VRF outputs are stored in the state storage or whole blocks are stored there, could you point me to the code in the runtime for this computation?